### PR TITLE
Delete own account

### LINKED ISSUE #763: add an option for deleting the own account

### PR DESCRIPTION
**Pull request by tobias-wiese**
# Draft until implemented on Lagom side
# Description

- Closes #763 

## Reason for this PR
- The user has to have an option to delete his account

## Changes in this PR
- a new section on the settings page, including a confirmation modal

## Type of change (remove all that don't apply)
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

**Test Configuration**:
- OS: Windows Mac Linux
- Browser: Firefox Chrome Safari Chromium

- Frontend: (remove all that don't apply)
  - [x] Development build
- Backend: (remove all that don't apply)
  - [ ] deployed development build
  - [ ] deployed experimental build

# Checklist: (remove all that don't apply)

- [x] I added corresponding E2E tests (especially for bugfixes)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new linting warnings or console warnings
- [x] I have updated the CHANGELOG.md to include any changes made in this PR (add WIP to the top, if there is none already)

# Screenshots:
grafik

grafik

### CODE REVIEW

File: src/components/modals/DeleteOwnAccountModal.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        <div v-else class="flex flex-col">
+            <div class="my-2 w-full flex justify-center my-8">
+                <div class="fa-stack fa-2x">
+                    <i class="fas fa-circle fa-stack-2x text-red-200" />
+                    <i class="fas fa-exclamation fa-stack-1x text-red-600" />
+                </div>
+            </div>
+            <p>
+                <span class="font-bold text-red-600">Warning!</span> You are about to delete your UC4 account. If you continue, all your
+                stored data will be deleted and you will not be able to use the system any more.
```
Comment (this-kramer) [inline src/components/modals/DeleteOwnAccountModal.vue:18]:
```suggestion
saved data will be permanently deleted and you will no longer be able to use the system. 
```

File: src/components/settings/DeleteAccountSection.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<template>
+    <section class="border-t-2 py-8 mt-5 border-red-400">
+        <div class="lg:flex">
+            <div class="w-full lg:w-1/3 lg:block mr-12 flex flex-col mb-4">
+                <div class="flex mb-2 align-baseline">
+                    <label class="block text-red-600 text-lg font-medium">Account Deletion</label>
+                </div>
+                <label class="block text-red-600"> Delete your account and get all stored data deleted. </label>
```
Comment (this-kramer) [inline src/components/settings/DeleteAccountSection.vue:8]:
```suggestion
                <label class="block text-red-600"> Delete my account and all my personal data. </label>
```
Maybe change this text to be more direct

**Review by this-kramer — CHANGES_REQUESTED** [review]
I think there are still some linter warnings that can be easily removed :smiley:

**Comment by tobias-wiese** [comment]
> 
> 
> I think there are still some linter warnings that can be easily removed 😃

The linting warnings are not caused by the changes on this branch. At least our linter configuration does not warn me about anything here (additionally pre-commit linting hook succeeded)

File: src/components/modals/DeleteOwnAccountModal.vue [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            <div class="my-2 w-full flex justify-center my-8">
+                <div class="fa-stack fa-2x">
+                    <i class="fas fa-circle fa-stack-2x text-red-200" />
+                    <i class="fas fa-exclamation fa-stack-1x text-red-600" />
+                </div>
+            </div>
+            <p>
+                <span class="font-bold text-red-600">Warning!</span>
+                You are about to delete your UC4 account. If you continue, all your saved data will be permanently deleted and you will no
+                longer be able to use the system.
```
Comment (bastihav) [inline src/components/modals/DeleteOwnAccountModal.vue:19]: image
Not sure how specific we should get, but all chain data remains and is still readable and (given the secret used at enrollment, which is currently called governmentId but could be anything) is also still connectable to a given person

**Review by bastihav — CHANGES_REQUESTED** [review]
Extremely aggresive and unreadable red, I would prefer a different shade.

File: src/components/modals/DeleteOwnAccountModal.vue (hunk 3 above)
Comment (tobias-wiese) [reply src/components/modals/DeleteOwnAccountModal.vue:19]: Maybe chaning it to "all your saved personal data..."?

Comment (tobias-wiese) [reply src/components/modals/DeleteOwnAccountModal.vue]: I change the shade of the red components from 500 to 700. Now it is a little friendlier to the eye. I also adjusted the describing text.
grafik

**Review by bastihav — APPROVED** [review]

**Review by this-kramer — APPROVED** [review]

**Pull request merged**